### DISCUSSION TITLE
Blazor Server: partially custom reconnection handler

### DISCUSSION THREAD

**Discussion by denisbredikhin** [dotnet/aspnetcore#29930] [Q&A]
Hello everyone.
I'm trying to find out what is the best way to handle reconnections in my app.
For my specific case default reconnection logic is almost ok, except that I would like to auto-reload a page in case when reconnection is rejected or failed.
And now we came to my problem: I can write fully custom reconnection logic, I can use standard one, or I can change the visualization of a standard one, but not to slightly change the standard logic.

As I see my problem can be solved in the following way:
- If it would be possible to pass not only reconnectCallback but also rejected and failed funcs to a DefaultReconnectionHandler. And if there will be documentation, how to initialize reconnectCallback with a DefaultReconnectionHandler.
- if I would be able to get prettified sources of  DefaultReconnectionHandler in js (then I can copy-paste it)
- if it would be able to customize Blazor startup in ts (as I know it is not possible?)
- if I would be able to pass custom descendant of DefaultReconnectDisplay to DefaultReconnectionHandler and initialize reconnectCallback  with it

Does someone know if anything from above is possible? Or maybe there is some other way to achieve what do I want?
P.S. I'm more a backend developer, so sorry for a possible mistakes in a frontend terminology

**Comment by mrlife** [comment]
I think you will need to create an issue for this or post somewhere like Stack Overflow.

See: https://github.com/dotnet/aspnetcore/issues/29935
